src/TraceEvent/Computers/StartStopActivityComputer.cs

```diff
@@ -957,53 +962,117 @@ void FixAndProcessAdoNetEvents(TraceEvent data)
         bool TryProcessDiagnosticSourceStartEvents(TraceEvent data)
         {
             Debug.Assert(data.ProviderGuid == MicrosoftDiagnosticsDiagnosticSourceProvider);
+            bool result = false;
             try
             {
                 string taskName = data.PayloadByName("EventName") as string;
                 if (taskName == null)
-                    return false;
-                string extraInfo = null;
-
-                // In the converter from 'DiagnosticSource' to ETW, we have some default configurations.
-                // By default, SQL (EntityFramework) DiagnosticSource event will be converted to 'Activity2Start' and 'Activity2Stop' events
-                // of provider 'Microsoft-Diagnostics-DiagnosticSource'. Meanwhile, the sql command will be in the payload.
-                // The following code is used to make the call stack looks better (converting 'Activity2Start' to 'SQLCommand')
-                // and showing the SQL command.
-                // Details:
-                //https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs
-
-                
-                var args = data.PayloadByName("Arguments") as DynamicTraceEventData.StructValue[];
-                if (args != null)
                 {
-                    var sb = Utilities.StringBuilderCache.Acquire(64);
-                    bool first = true;
-                    foreach (var arg in args)
+                    result = false;
+                }
+                else
+                {
+                    string extraInfo = null;
+                    switch (taskName)
                     {
-                        string key = arg["Key"] as string;
-                        object value = arg["Value"] as string;
-                        if (key != null && value != null)
-                        {
-                            string valueStr = value.ToString();
-                            if (!string.IsNullOrEmpty(valueStr))
+                        case "Microsoft.EntityFrameworkCore.BeforeExecuteCommand":
+                            // In the converter from 'DiagnosticSource' to ETW, we have some default configurations.
+                            // By default, SQL (EntityFramework) DiagnosticSource event will be converted to 'Activity2Start' and 'Activity2Stop' events
+                            // of provider 'Microsoft-Diagnostics-DiagnosticSource'. Meanwhile, the sql command will be in the payload.
+                            // The following code is used to make the call stack looks better (converting 'Activity2Start' to 'SQLCommand')
+                            // and showing the SQL command.
+                            // Details:
+                            //https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs
+
```

Comment:
Is this comment line really not indented?   If so go ahead an indent it.  It is OK if the line is long.

Reply:
Bad view from github review page. The comment is indented well.